lib/bolt/transport/ssh/connection.rb

```diff
@@ -242,7 +242,11 @@ def net_ssh_verifier(verifier)
         end
 
         def shell
-          @shell ||= Bolt::Shell::Bash.new(target, self)
+          @shell ||= if target.options['login-shell'] == 'powershell'
+                       Bolt::Shell::Powershell.new(target, self)
+                     else
+                       Bolt::Shell::Bash.new(target, self)
```

Comment:
I wonder if 'unix' makes more sense for this now? I was thinking of putting up a 'cleanup' PR - I think I missed a few "remote" var names in the first PR, and want to reassess unifying the transport classes that just define `with_connection`. Renaming this could be part of that.